### PR TITLE
[Templates] Fix for QA content model change on article listings pages for Resources & Support

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR implements these changes to the "article listings" pages for the Learning Center (Resources and Support.)

- [x] Update template based on content model change (follow up to https://github.com/department-of-veterans-affairs/vets-website/pull/14639)
    - This fixes the snippet below each article.
- [x] Use the Drupal-supplied URL for listings instead of generating it
    - I didn't know the Drupal content model includes an entityUrl for each category or tag. This PR incorporates that field.
- [x] Add the section-wide search bar

https://github.com/department-of-veterans-affairs/va.gov-team/issues/14841
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14835

## Testing done
Observed and tested the page

## Screenshots
Notice the URL and the full snippet for each article.

image

### Search bar

image

## Acceptance criteria
- [ ] Article listings has the right URL and shows article snippets

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/drupal/graphql/taxonomy-fragments/termLcCategory.taxonomy.graphql.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 module.exports = `
 fragment taxonomyTermLcCategories on TaxonomyTermLcCategories {
+  entityUrl {
+    path
+  }
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/taxonomy-fragments/termLcCategory.taxonomy.graphql.js:5]: This is the property indicating the URL for a category listing.

File: src/site/layouts/support_resources_article_listing.drupal.liquid [hunk 2]
```diff
@@ -21,7 +21,7 @@
                 {% assign snippet = article.fieldIntroTextLimitedHtml %}
 
                 {% if article.entityBundle == 'q_a' %}
-                  {% assign snippet = article.fieldAnswer.0.entity.fieldWysiwyg %}
+                  {% assign snippet = article.fieldAnswer.entity.fieldWysiwyg %}
```
Comment (ncksllvn) [inline src/site/layouts/support_resources_article_listing.drupal.liquid:24]: This is the fix for the article snippets not showing for QA pages.

File: src/site/stages/build/plugins/create-resources-and-support-section.js [hunk 3]
```diff
@@ -47,56 +47,76 @@ function groupByTags(allArticles) {
...
+
+    const primaryCategoryName = fieldPrimaryCategory.entity.name;
+    const primaryCategoryArticleListing =
+      articleListingsByCategory[primaryCategoryName];
+
+    if (!primaryCategoryArticleListing) {
+      articleListingsByCategory[primaryCategoryName] = {
+        articles: [article],
+        entityUrl: fieldPrimaryCategory.entity.entityUrl,
+      };
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-resources-and-support-section.js:82]: This is where the Drupal "entityUrl" is taken into account during the build for article listings by category

File: src/site/layouts/support_resources_article_listing.drupal.liquid (hunk 2 above)
Comment (kelsonic) [reply src/site/layouts/support_resources_article_listing.drupal.liquid:24]: I have a very random liquid template question:

Let's say I have an array called items (`[1, 2, 3]`) and another array called otherItems (`[4, 5, 6]`) and I want to do something like this:

```liquid
{% for item in items %}
  {% assign itemIndex = forloop.index %}

  {% for otherItem in otherItems %}
    {{ item[itemIndex] }}
  {% endfor %}
{% endfor %}
```

I'm just trying to figure out if you can do `{{ array[forloop.index] }}` in liquid, @youngfreezyVA and I weren't able to get that to work

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

File: src/site/includes/support_resources_search_bar.drupal.liquid [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        form.action = '/search';
+      }
+
+      // Prevent the "path" form field from being included in the form data to keep the URL cleaner.
+      // That field is only used to decide where to submit the search.
+      searchGlobally.removeAttribute('name');
+      document.getElementById('search-within-resources-and-support').removeAttribute('name');
+    });
+  })();
+</script>
```
Comment (ncksllvn) [inline src/site/includes/support_resources_search_bar.drupal.liquid:50]: Takes me back...

File: src/site/includes/support_resources_search_bar.drupal.liquid [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      </div>
+    </fieldset>
+    <label class="vads-u-margin-top--1" for="query">Enter a keyword, phrase, or question</label>
+    <div class="vads-l-row">
+      <div class="vads-l-col--12 vads-u-flex--1 vads-u-width--auto">
+        <input type="text" required name="query" id="query" class="usa-input vads-u-max-width--100 vads-u-width--full vads-u-height--full vads-u-margin--0">
+      </div>
+      <div class="vads-l-col--12 vads-u-flex--auto vads-u-width--auto">
+        <button type="submit" class="usa-button vads-u-margin--0 vads-u-width--full vads-u-width--auto vads-u-height--full" style="border-top-left-radius: 0; border-bottom-left-radius: 0;">
+          <i class="fa fa-search"> </i> Search
```
[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/site/includes/support_resources_search_bar.drupal.liquid:28]: Design spec 
image

**Review by rianfowler — APPROVED** [review]

File: src/site/includes/support_resources_search_bar.drupal.liquid (hunk 4 above)
Comment (youngfreezyVA) [reply src/site/includes/support_resources_search_bar.drupal.liquid:50]: oh snap!! good to know you can do this in liquid templates

Comment (ncksllvn) [reply src/site/includes/support_resources_search_bar.drupal.liquid:50]: It's probably not something we should do constantly but I think it's fine for little snippets like this

**Pull request merged**